pkg/termination/termination.go

```diff
@@ -90,6 +91,28 @@ func (h *handler) Run(stop <-chan struct{}) error {
 }
 
 func (h *handler) run(ctx context.Context) error {
+	httpClient := http.Client{
+		// Copied from the http.DefaultTransport
+		// https://github.com/golang/go/blob/f92337422ef2ca27464c198bb3426d2dc4661653/src/net/http/transport.go#L42-L54
+		Transport: &http.Transport{
+			Proxy: http.ProxyFromEnvironment,
+			DialContext: (&net.Dialer{
+				Timeout:   30 * time.Second,
+				KeepAlive: 30 * time.Second,
+				DualStack: true,
+			}).DialContext,
+			ForceAttemptHTTP2:     true,
+			MaxIdleConns:          100,
+			IdleConnTimeout:       90 * time.Second,
+			TLSHandshakeTimeout:   10 * time.Second,
+			ExpectContinueTimeout: 1 * time.Second,
+
+			// By disabling keep alives, this allows us to intercept metadata traffic during E2E testing
+			// and allows us to simulate a termination event.
+			DisableKeepAlives: true,
```

Comment:
This seems like a large compromise just to make tests work.

Can you link to what we're doing in the e2e tests for reference?

Reply:
> This seems like a large compromise just to make tests work.

Agreed, though I can't really think of a better way to allow this test to work. I've tried killing the connection using `ss` from the privileged container produced by the test, but to no avail. We could disable the test/delete it, but that's also not great.

For reference it is the test that was introduced in: https://github.com/openshift/cluster-api-actuator-pkg/pull/171, do you think this should be linked in the code somewhere?